### DISCUSSION TITLE
Formato de los datos (lista de estudios con su información)

### DISCUSSION THREAD

**Discussion by AlainPlusPlus** [indi-es/estudios#22] [General]
### JSON vs XML

¿Utilizamos **[JSON o XML](https://www.w3schools.com/js/js_json_xml.asp)** para acceder los datos desde el front end?
No creo que sea necesario una base de datos por el momento.

**Comment by afk-mario** [comment]
Por ahorta estaba usando Markdown, porque siento que es lo más fácil para que la gente pueda aportar, aquí dejé un script que extrae todos los links y prueba a vert si alguno anda caido.

https://github.com/indi-es/estudios/blob/main/.github/site-checker/index.js

**Comment by AlainPlusPlus** [comment]
¡Perfecto! Me imagino que si se llega a hacer la web, Markdown estará bien para pasar los datos a HTML y luego darle formato con CCS. Igual se puede cambiar el formato en el futuro si se necesita.

¿Para contribuir con la lista lo hacemos en [main,](https://github.com/indi-es/estudios) directamente, cierto?

**Comment by afk-mario** [comment]
Si a main está perfecto.

Algunas preguntas que no se bien cuales son las respuestas todavía.

### Deberiamos guardar más información que el nombre y link a su página web?

Creo que si agregamos más datos serían cosas que no cambién mucho, por ejemplo no agregaría el numero de empleados porque es algo que se tiene que estar actualizando constantemente, pero tal vez valga la pena tener una lista de juegos?

### Categorías

En el discord e INDI.ES hemos invitado gente de todos lados, el único requisito es hablar español, supongo que por aquí es lo mismo, no se si valdría la pena dividir por estados a los estudios.

### Format

Si agregamos más datos tal vez valga la pena guardar la infomaración en otra estructura (JSON, YAML, etc) y con un Github Action generar un markdown
### Solo devs o freelance

Agregamos devs que son ellos solos ? y de alguna manera los marcamos como diferentes ?

**Comment by Ucamo** [comment]
Apoyo la idea de la lista de juegos, y estoy de acuerdo conque si vamos agregando mas datos quizás el markdown nos quede corto. 

Para juegos quizás algo como:
-Imagen
-Link a su store
-Descripción
-Estudio/Author

Pero ya caemos mucho en estar haciendo el scraping nosotros, podríamos poner un form donde nos puedan llenar esos datos y si quieren ser incluídos que los llenen?, igual no nos ahorrariamos el andar buscando nosotros solos, pero nos ahorraría algo de trabajo manual si tuvieramos un lugar en el que otros pudieran colaborar llenando esos datos.

Si incluyeramos juegos, los solo devs y freelance podrían entrar ahí con sus propios juegos bajo "Estudio/Author"

**Comment by AlainPlusPlus** [comment]
### Deberiamos guardar más información que el nombre y link a su página web?
La lista de juegos estaría bien (junto con lo que menciona @Ucamo) pero no se si sería útil o cual sería el proposito en este momento. Igual lo podíamos agregar más adelante si se necesita. Utilizar formularios para automatizar el proceso sería lo ideal y habría que ver como nos encargaríamos de hacer la moderación/validación.
Por ahora, nos podríamos enfocar en lo más importante: **nombre de estudios y sus página webs**. Más adelante podríamos pedir por Twitter que cada estudio llene los datos con sus juegos y así darles a ellos la responsabilidad de ir actualizando eso. 

### Categorías
Buena idea pensar en categorías (estados/países). Creo que por lo pronto nos enfocamos sólo en México y en el futuro agregamos más países, si esto le serviria a los que usen la lista. De los estados, no creo que sea tan fácil obtener esa información fácilmente en estos momentos pero si valdría la pena conseguirla más adelante.

### Formato
Suena bien la idea de @afk-mario para un futuro cercano (tomando en cuenta lo que comenta @Ucamo). Cualquiera de las opciones (JSON/YAML/ XML) dependería del lenguaje/herramienta que usemos para presentar los dados en la página web y en Github generamos el markdown para fácil acceso.

### Solo devs o freelance
Aquí si delegaría esto totalmente a los Devs/Freelance que quieran estar en la lista por que son bastantes. No hay problema con agregarlos pero, ¿es algo de lo que la comunidad se beneficiaria? Creo que los reclutadores/reclutadoras son los que más se beneficiarían.

**Comment by afk-mario** [comment]
### Información
Estube pensando en los juegos, creo que aplica lo mismo que las redes sociales, esa información debe de estar en cada página de los estudios y creo que los mejores para tenerla actualizada son los mismos, si nosotros nos damos a la tarea de mantener esa información puede llegar a ser mucha chamba y duplicada.

**Comment by afk-mario** [comment]
### Recopilación de información

No se si me encanta la idea del Form, una vez que alguien submitea información es difícil darle feedback sobre la información que dió, tal vez fue duplicada, tal vez falto algo que necesitabamos o no sigue con los lineamientos que pongamos, en GH podemos abrir una discucución con cada nueva entrada de info, el problema de GH es que es menos accesible que una forma de Google, pero creo que la mayoria de los que van querer tener esta información recopilada serían estudios e idealmente tendrán por lo menos un programador que los oriente a usar GH, también podemos mitigarlo creando una guía para las demás personas

**Comment by NikanCode** [comment]
Entonces por lo que entiendo de momento sería solo recopilar la lista de estudios lo más completa y amplia posible y ya a partir de eso ver la forma de mantenerla actualizada quitando y agregando para dejar en todo momento una lista de enlaces activos y ya después se vería si se quiere abarcar más.

En el sentido de los enlaces por redes sociales lo pensé principalmente para evitar que el sitio estuviera no disponible en la lista, pero supongo que el enlace se quitaría si no esta activo.

**Comment by AlainPlusPlus** [comment]
### Información
¿Esta bien si agregamos el link de la red social (Twitter supongo) para los estudios que no tienen página y que estan activos como se platicó en el Discord? 
### Solo devs o freelance
¿Les parece si para los Devs/Freelance los agregamos en un archivo aparte o los empezamos a agregar ya a la lista principal de los estudios? Por lo menos a los que ya tenemos del Discord o los del Tweet the Fayer, por ejemplo. Y a cuando tengamos solucionado lo de las instrucciones o una forma de que ellos mismos puedan agregar sus datos les avisamos a todos.

**Comment by afk-mario** [comment]
Estaba pensando en la parte de juegos, podría funcionar tener una sección de *showcase* que sería para juegos que han salido de miembros del Discord, y que han publicado en #muestra-pregunta.

**Comment by afk-mario** [comment]
> ### Información
> 
> ¿Esta bien si agregamos el link de la red social (Twitter supongo) para los estudios que no tienen página y que estan activos como se platicó en el Discord?
> ### Solo devs o freelance
> 
> ¿Les parece si para los Devs/Freelance los agregamos en un archivo aparte o los empezamos a agregar ya a la lista principal de los estudios? Por lo menos a los que ya tenemos del Discord o los del Tweet the Fayer, por ejemplo. Y a cuando tengamos solucionado lo de las instrucciones o una forma de que ellos mismos puedan agregar sus datos les avisamos a todos.

Sip, creo que agregar un Link por estudio está bien y podemos tener esta preferencia

Página Web -> Página web de juego -> Red social del estudio -> Red social del juego.

La primera que tengan es la que agregamos.

**Comment by AlainPlusPlus** [comment]
Empecé con la lista de desarolladores/freelancers para tenerla como ejemplo para que después se vayan agredando más. Después vemos como los agregamos a la lista principal.
PR: https://github.com/indi-es/estudios/pull/10

**Comment by AlainPlusPlus** [comment]
### Estudios de mexicanos en el extranjero

Otra lista. Ahora hay que ver este si tomamos en cuenta este otro caso (edge case) de estudios en el extranjero donde mexicanos estan haciendo juegos. ¿Qúe opinan?
PR: https://github.com/indi-es/estudios/pull/13

**Comment by afk-mario** [comment]
Creo que está válido :)

**Comment by AlainPlusPlus** [comment]
### Esquema
Propuesta del esquema (quitando cosas básicas como nombre, ID, etc...) de como podría quedar todo organizado hasta lo que llevamos ahora.

**Estudio**:
- Fundador/creador
- Año que se fundó (oficialmente?)
- País
- Estado (*si esta en México)
- Juegos
- Website

**Desarollador/freelancer**:
- País
- Estado (*si esta en México)
- Juegos
- Website (red social)

**Juegos** _(basado en lo que propuso @Ucamo)_:
- Imagen
- Estudio/Author
- Descripción
- Plataformas (con link a su store)
- Website

_[Aquí](https://younghorsesgames.com/press/sheet.php?p=octodad_dadliest_catch) un ejemplo supercompleto (onda press kit) del cual nos podríamos basar la parte de juegos._

**Comment by afk-mario** [comment]
> **Estudio**:
> 
>     * Fundador/creador

Lo pensaría mas como persona responsable? como contacto/CEO o algo por el estilo aunque no se que tan ambiguo se vuelva esto y si mejor no lo ponemos. 

> 
>     * Año que se fundó (oficialmente?)

Siento que no es tan relevante

>     * Juegos

Yo estoy un poco en contra de tener la lista de juegos, o lo podria como paso 3 (creo que agregar la info nueva seria el 2)
 
>     * Website
> 

> **Juegos** _(basado en lo que propuso @Ucamo)_:
> 
>     * Imagen
> 
>     * Estudio/Author
> 
>     * Descripción
> 
>     * Plataformas (con link a su store)
> 
>     * Website
> _[Aquí](https://younghorsesgames.com/press/sheet.php?p=octodad_dadliest_catch) un ejemplo supercompleto (onda press kit) del cual nos podríamos basar la parte de juegos._

Me sigue gustando la idea de que fueran los showcases de la comunidad de INDI.ES pero creo que si si se quiere tener una lista completa de todos los juegos que han salido por devs hispanoparlantes puede ser una camba con un scope ya muy alto a lo que estamos haciendo, igual creo que lo pondría como paso aparte.

Creo que para cualquier buen listado en una página está bien una imagen que ayude a relacionar el estudio idealmente sería el logo en SVG pero pues esto puede ser difícil, aunque estaba pensando que esto puede ser el lineamiento de si un estudio entra o no entra a la lista?

- Tiene página web (maybe seguimos con página de red social activa, fb, tw, itch)
- Tiene logo en SVG

Nos estamos acercando a los 100 estudios y tal vez vale la pena tener ese filtro por muy pequeño que fuera

**Comment by Ucamo** [comment]
A lo mejor incluir las imagenes para esta versión si se puede salir de las manos (por cuestiones de andar correteando logos, o que no todos serán de la misma medida y así). Me gusta el progreso que llevamos, quizás podríamos agregar algo como
"No encuentras el estudio que buscas/Quisieras que se incluyera tu estudio a esta lista?" y ponemos un enlace a.... algo donde podamos recolectar los datos y que nos ayude a centralizar un lugar donde podamos ver los estudios pendientes de agregar e ir agregandolos para las personas que quieran colaborar (dentro del estudio o incluso sus fans) y que no le sepan al proceso de git.

(o... hacer un tutorial de como hacer un PR?)

en otras cosas, como vamos a saber que tenemos lista la primera versión? O:

**Comment by NikanCode** [comment]
Creo que estaría bien hacer lo del pequeño tutorial para que agregues tu solicitud de agregarte a una de las listas que ya hay actualmente.

¿deberíamos ya cambiar el nombre de Readme.md, por estudios.md o algo así?

**Comment by afk-mario** [comment]
> (o... hacer un tutorial de como hacer un PR?)

https://docs.github.com/en/github/building-a-strong-community/setting-guidelines-for-repository-contributors

Creo que podemos empezar con un tutorial ya sea hosteado aparte o en el mismo repo con un CONTRIBUTING.md linkear el repo y si vemos que si hay mucha fricción al agregar estudios empezara a trabajar en una alternative (google forms o algo por el estilo)

> 

Creo que ya valdría la pena mover la info a sus respectivos archivos, y maybe usar un GH action para hacer de (JSON, YAML, TOML, etc) -> Markdown 


Me lo imagino como 

```
|
|- estudios-mexico.json
|- estudios-fuera-de-mexico.json
|- README.md

```

**Comment by AlainPlusPlus** [comment]
> Lo pensaría mas como persona responsable? como contacto/CEO o algo por el estilo aunque no se que tan ambiguo se vuelva esto y si mejor no lo ponemos.

Lo podemos omitir.

> Yo estoy un poco en contra de tener la lista de juegos, o lo podria como paso 3 (creo que agregar la info nueva seria el 2)

Creo que lo podríamos dejar opcional para cuando les digamos a los estudios que agreguen más de su propia información (Si vamos a ir agregando Estados, por ejemplo.)

> Me sigue gustando la idea de que fueran los showcases de la comunidad de INDI.ES pero creo que si si se quiere tener una lista completa de todos los juegos que han salido por devs hispanoparlantes puede ser una camba con un scope ya muy alto a lo que estamos haciendo, igual creo que lo pondría como paso aparte.

Cierto, este showcase más elaborado se puede hacer mejor con los la comunidad de INDI.ES.
Mi idea original de la lista de juegos era una lista donde se mostraran todos los juegos que se han hecho (y publicado) en México, aunque sea de nombre pero esto se podría hacer en el paso 2, cuando agregamos más información con ayuda de los estudios/developers/freelancers. 

> Creo que para cualquier buen listado en una página está bien una imagen que ayude a relacionar el estudio idealmente sería el logo en SVG pero pues esto puede ser difícil, aunque estaba pensando que esto puede ser el lineamiento de si un estudio entra o no entra a la lista?
> 
>     * Tiene página web (maybe seguimos con página de red social activa, fb, tw, itch)
> 
>     * Tiene logo en SVG
> 
> 
> Nos estamos acercando a los 100 estudios y tal vez vale la pena tener ese filtro por muy pequeño que fuera

Me parece bien tener este filtro pero entonces, ¿el logo en SVG se lo tendríamos que pedir a los estudios en el paso 2, cierto?

**Comment by AlainPlusPlus** [comment]
> en otras cosas, como vamos a saber que tenemos lista la primera versión? 

Así es como tengo visualicé los pasos del plan pero corrijanme si me equivoco:

1. La primera versión ya casi la tenemos, que era hacer la lista de estudios mexicanos con los links a sus páginas/redes sociales. Ahora lo que faltaría es trabajar en el contributing y organizar bien los datos que tenemos en su respectivo formato (como lo comenta @afk-mario en su último mensaje).
Si ponemos a los desarrollador-estudios (gente que hace/publica juegos con su nombre/marca como Calix, Omael, Mario, etc) en la lista de estudios, la otra lista de desarolladores/freelancers podría ser para gente que no ha publicado nada formal y que podrían ser de varias especialidades (no solo developers), por ejemplo. Así dejamps a algunos que ya estan ahí como ejemplo para que los que les interese contribuir lo hagan por medio de un PR/Form (según las instrucciones).

2. El segundo paso sería ya contactar a los **estudios, freelancers y miebros de INDI.ES** para obtener más información y que ellos nos la pasen. Aquí las instrucciones en el **CONTRIBUTING.md** van a ser importantes.

3. Aceptar PRs y ver qué más cosas se podrían agregar/mejorar.

**Yo no voy a tener mucho tiempo el fin de semana pero podría ayudar a organizar y a generar los archivos (JSON) o a investigar lo del GH actions.**

**Comment by afk-mario** [comment]
> Me parece bien tener este filtro pero entonces, ¿el logo en SVG se lo tendríamos que pedir a los estudios en el paso 2, cierto?

He estado pensando en esto y creo que el logo en SVG si es mucho pedir :/ pero creo que con que este en buena resolución y png/svg estaría bien.

**Comment by AlainPlusPlus** [comment]
Poner SVG como formato de preferencia esta perfecto. No se como se ha pensado el diseño de la presentación final de la página de los estudios.

**Comment by AlainPlusPlus** [comment]
Ya esta el pull request de los archivo con los estudios en formato JSON. El formato es el que ya habíamos acordado pero lo puedo modificar si se me pasó algo. Al rato le doy al script para pasar de **JSON a Markdown**. Voy a generar un archivo intermediario para no cambirar el **README.md.**

[Aquí](https://github.com/microsoft/vscode/blob/master/CONTRIBUTING.md) un ejemplo del que nos podríamos basar para el **CONTRIBUTING.md** 📖 . @NikanCode y yo hablamos un poco de esto en el chat de voz del discord hace unos días.

**Comment by AlainPlusPlus** [comment]
Otro [pull request](https://github.com/indi-es/estudios/pull/15) relacionado a pasar los datos de **JSON a Markdown**. A ver si alquien tiene chance de checarla cuando puedan.

@afk-mario, tengo preguntas acerca de la web antes de implementar el GH action para automatizar esto:
- ¿Queremos el script modifique el README? me imagino que sólo se tendría que cambiar la lista y dejar todo lo demás intacto pero no se si vale la pena hacerlo ya ahora si habrá un cambios en como se muestra la lista en la página de indi-es.
- ¿Tenemos pensado cambiar el formato de la lista para la versión final de la web?
- ¿Sería buena idea ir pensando en la implementación de una BD y/o el backend?

Independiente de mis preguntas, creo que por lo pronto valdia la pena que el link de **Estudios** en indi-es.com mandara [aquí](https://github.com/indi-es/estudios/blob/main/README.md) en lugar de la página principal del proyecto en Github.

**Comment by AlainPlusPlus** [comment]
Recuerdo que se comentó el uso de la API de Notion. Encontré [esto](https://github.com/jamalex/notion-py) pero no es oficial. 🤔

**Comment by AlainPlusPlus** [comment]
Recopilar más información de los estudios (Form/Github): https://airtable.com/integrations/github

**Comment by AlainPlusPlus** [comment]
# RESUMEN
_(Llamada del 21.02.2021)_

### Formato (JSON y Markdown)
Se decidió utilizar [JSON](https://en.wikipedia.org/wiki/JSON) para la serialización de datos y de ahí generar el [Markdown](https://en.wikipedia.org/wiki/Markdown) para fácil visualización/acceso en el repo.
[YAML](https://en.wikipedia.org/wiki/YAML#Comparison_with_JSON), TOML and XML también fueron mencionados pero el formato JSON es relativamente fácil de leer y también de modificar. YAML es más difícil de "parcear" y menos usado; Como es un superset de JSON, algunos "parcers" de YAML aceptan JSON.

### JSON (Campos acordados hasta [ahora]( https://github.com/indi-es/estudios/issues/19))
```
"Estudios": [
    {
      nombre: 'Mecha Studios',
      pais: 'México',
      estado: '',
      ciudad: '',
      tipo: 'Estudio',
      activo: true,
      ultima_vez_activo: '2021-02-21T23:54:02.104Z',
      twitter: '',
      facebook: '',
      instagram: '',
      website: 'https://www.mechastudios.com/'
    }
]
```

### Estructura de archivos
```
|
|- estudios-mexico.json
|- estudios-fuera-de-mexico.json
|- README.md

```

### Recopilar más información de los estudios
Manual:
- https://github.com/indi-es/estudios/issues/21
- https://github.com/indi-es/estudios/issues/20

Automatico:
- https://github.com/indi-es/estudios/discussions/23

### Web
Esta hecha en [Next.js](https://nextjs.org/) y para mostrar los estudios nos basaremos en [estos ejemplos](https://github.com/indi-es/estudios/discussions/24). Los estudios/datos se importarán del JSON.

### Contribuir al proyecto
[Aquí](https://github.com/indi-es/estudios/issues/18) está el issue para lo del documento de cómo contribuir.
